src/pages/trader-tools/common/_utility.js

```diff
@@ -52,13 +53,136 @@ const getSwapChargeForex = (values) => {
     return toFixed(swap_formula_forex)
 }
 
+// PnL Margin Calculator
+const getStopLossLevel = (values) => {
+    let { assetPrice, stopLossAmount, volume, contractSize } = values
+    assetPrice = Number(assetPrice)
+    stopLossAmount = Number(stopLossAmount)
+    const stop_loss_level_formula = assetPrice + stopLossAmount / (volume * contractSize)
+    return toFixed(stop_loss_level_formula)
+}
+
+const getTakeProfitLevel = (values) => {
+    let { assetPrice, takeProfitAmount, volume, contractSize } = values
+    assetPrice = Number(assetPrice)
+    takeProfitAmount = Number(takeProfitAmount)
+    const take_profit_level_formula = assetPrice - takeProfitAmount / (volume * contractSize)
+    return toFixed(take_profit_level_formula)
+}
+
+const getStopLossPip = (values) => {
+    let { assetPrice, stopLossLevel, pointValue } = values
+    assetPrice = Number(assetPrice)
+    stopLossLevel = Number(stopLossLevel)
+    const stop_loss_pip_formula = Math.abs(stopLossLevel - assetPrice) / pointValue
+    return toFixed(stop_loss_pip_formula)
+}
+
+const getTakeProfitPip = (values) => {
+    let { assetPrice, takeProfitLevel, pointValue } = values
+    assetPrice = Number(assetPrice)
+    takeProfitLevel = Number(takeProfitLevel)
+    const take_profit_pip_formula = Math.abs(takeProfitLevel - assetPrice) / pointValue
+    return toFixed(take_profit_pip_formula)
+}
+
+// PnL Multipliers Calculator
+const getStopLossLevelUp = (values) => {
```

Comment:
All the PnL functions has a lot in common, can we somehow create a new function with the common codes and use it multiple times?